framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -894,7 +892,11 @@ public AnnotationFormatter getAnnotationFormatter() {
      *         empty set if no qualifiers are supported
      */
     public final Set<Class<? extends Annotation>> getSupportedTypeQualifiers() {
-        return supportedQuals;
+        if (this.supportedQuals.isEmpty()) {
+            supportedQuals.addAll(createSupportedTypeQualifiers());
```

Comment:
`addAll()` reads whatever is returned by `createSupportedTypeQualifiers()`, essentially copying all of the annotation class objects into `supportedQuals`. I don't see why you've eliminated the contract that `createSupportedTypeQualifiers` returns an **immutable** set of qualifiers. Keeping the contract ensures that even during the creation of this set, we eliminate any chances of unwanted modification of the set of supported qualifiers.